protostar/cli/signer_factory.py

```diff
@@ -0,0 +1,97 @@
+import importlib
+import os
+from pathlib import Path
+from typing import Optional
+
+from starknet_py.net.models import StarknetChainId
+from starknet_py.net.signer import BaseSigner
+
+from protostar.cli import Command
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet_gateway import AccountConfig, create_stark_curve_signer
+
+PRIVATE_KEY_ENV_VAR_NAME = "PROTOSTAR_ACCOUNT_PRIVATE_KEY"
+
+
+SIGNABLE_ARGUMENTS = [
+    Command.Argument(
+        name="account-address",
+        description="Account address",
+        type="str",
+    ),
+    Command.Argument(
+        name="private-key-path",
+        description="Path to the file, which stores your private key (in hex representation) for the account. \n"
+        f"Can be used instead of {PRIVATE_KEY_ENV_VAR_NAME} env variable.",
+        type="path",
+    ),
+    Command.Argument(
+        name="signer-class",
+        description="Custom signer class module path.",
+        type="str",
+    ),
+]
+
+
+def create_account_config_from_args(
+    args, chain_id: StarknetChainId
+) -> Optional[AccountConfig]:
+    custom_signer: Optional[BaseSigner] = None
+    if args.account_address is None:
+        return None
+    if args.signer_class:
+        custom_signer = create_custom_signer(args.signer_class)
+        return AccountConfig(account_address=args.account_address, signer=custom_signer)
+    return AccountConfig(
+        account_address=args.account_address,
+        signer=create_stark_curve_signer(
+            account_address=args.account_address,
+            private_key=get_private_key(args.private_key_path),
+            chain_id=chain_id,
+        ),
+    )
+
+
+def create_custom_signer(signer_class_path: str) -> BaseSigner:
+    *module_names, class_name = signer_class_path.split(".")
```

Comment:
no need for immediate action: I recall somewhere that a convention for specifying Python module items looks like this: `module.path.a.b.c:item_name`. Gunicorn does this for example: https://gunicorn.org/